spec/formatting.md

```diff
@@ -8,6 +8,15 @@ when formatting a message for display in a user interface, or for some later pro
 The document is part of the MessageFormat 2.0 specification,
 the successor to ICU MessageFormat, henceforth called ICU MessageFormat 1.0.
 
+## Literal Resolution
+
+The resolved value of _text_, _literal_ and _nmtoken_ tokens
+is always a string concatenation of its parts,
+with escape sequences resolving to their escaped characters.
+When a literal value is used as a formatting function argument or option value,
+the formatting function MUST treat option values the same independently of their presentation,
```

Comment:
Going a bit further, I sensed on our call yesterday for us to have a consensus on formatting functions not being able to determine if an option has been set from a literal or variable. As in, a formatting function receiving a value `'42'` for the `foo` option would not know if the message had set `foo=42`, `foo=|42|`, or `foo=$bar` where `$bar` had a value of `'42'`.

Reply:
Then, given the consensus, I think it would make sense to rephrase this as a requirement on *whatever calls the functions* (the formatter?), not on the functions itself. It's true that the formatting function must treat option values the same independently of their presentation, but it's also impossible for it to do otherwise!

A way to phrase it might be that the domain of the formatting function is *resolved values*, and then if you wanted to add examples, one example could be that `42`, `|42|`, and `$bar` all map to the same resolved value (assuming an environment in which `bar` is bound to 42), specifically `'42'`.